src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1150,6 +1192,45 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         args.Handled(true);
     }
 
+    // Method Description:
+    // - Event handler for the PointerEntered event. We use this for cursor manipulation.
+    // Arguments:
+    // - sender: the XAML element responding to the pointer input
+    // - args: event data
+    void TermControl::_PointerEnteredHandler(Windows::Foundation::IInspectable const& /*sender*/,
+                                             Input::PointerRoutedEventArgs const& /*args*/)
+    {
+        if (_closing)
+        {
+            return;
+        }
+
+        auto coreWindow = Window::Current().CoreWindow();
+        _oldCursor = coreWindow.PointerCursor();
+
+        if (_terminal->IsTrackingMouseInput())
+        {
+            return;
+        }
+
+        coreWindow.PointerCursor(_textCursor);

```

Comment:
What if we've got two SxS panes, press shift in one, then hover over the other?

Reply:
That works fine. The cursor is updated when the pointer enters and leaves the control.